core/network/src/protocol.rs

```diff
@@ -705,12 +706,11 @@ impl<B: BlockT, S: NetworkSpecialization<B>, H: ExHashT> Protocol<B, S, H> {
 		let mut peers = self.context_data.peers.write();
 		let hash = header.hash();
 		for (who, ref mut peer) in peers.iter_mut() {
-			if peer.known_blocks.insert(hash.clone()) {
-				trace!(target: "sync", "Reannouncing block {:?} to {}", hash, who);
-				self.send_message(io, *who, GenericMessage::BlockAnnounce(message::BlockAnnounce {
-					header: header.clone()
-				}));
-			}
+			peer.known_blocks.insert(hash.clone());
```

Comment:
Why do we need to clone this hash? Couldn't we just put the trace one line above?

Reply:
it definitely needs to be cloned because this is in a loop. the trace is not relevant to that